src/reanimated2/initializers.ts

```diff
@@ -0,0 +1,104 @@
+import { reportFatalErrorOnJS } from './errors';
+import NativeReanimatedModule from './NativeReanimated';
+import { runOnUI, runOnJS } from './threads';
+
+function callGuardDEV<T extends Array<any>, U>(
+  fn: (...args: T) => U,
+  ...args: T
+): void {
+  'worklet';
+  try {
+    fn(...args);
+  } catch (e) {
+    if (global.ErrorUtils) {
+      global.ErrorUtils.reportFatalError(e as Error);
+    } else {
+      throw e;
+    }
+  }
+}
+
+function callGuard<T extends Array<any>, U>(
+  fn: (...args: T) => U,
+  ...args: T
+): void {
+  'worklet';
+  fn(...args);
+}
+
+function valueUnpacker(objectToUnpack: any, category?: string): any {
+  'worklet';
+  let workletsCache = global.__workletsCache;
+  let handleCache = global.__handleCache;
+  if (workletsCache === undefined) {
+    // init
+    workletsCache = global.__workletsCache = new Map();
+    handleCache = global.__handleCache = new WeakMap();
```

Comment:
I'm confused, they are already in initializers.ts – maybe you mean to move them to `initializeUIRuntime` method? This could be done but I'd rather do it in the future as it requires a few more things to be cleaned up. The main reason is that `valueUnpacker` is actually used to facilitate `runOnUI` that is used to schedule `initializeUIRuntime`. Hence this method is used before `intiializeUIRuntime` and so we won't be able to have these objects ready.

Reply:
Yeah, `initializeUIRuntime` is what I meant. Thanks for explanation